packages/react/src/patterns/sub-patterns/TableOfContents/TableOfContents.js

```diff
@@ -33,34 +31,31 @@ const { prefix } = settings;
  */
 const TableOfContents = ({ menuItems, children, menuLabel, theme }) => {
   const [selectedId, setSelectedId] = useState(menuItems[0].id);
-  const [selectedTitle, setSelectedTitle] = useState(menuItems[0].title);
-
+  const [selectedTitle, setSelectedTitle] = useState(menuItems[0].name);
+  const useMenuItems = menuItems ? [...menuItems] : findMenuItems();
   useEffect(() => {
     scrollStop(setSelectedItem);
   });
 
+  if (!useMenuItems) {
+    findMenuItems();
+  }
+
   /**
-   * Check whether provided anchor tags are in visible viewport
+   * loops into the array of elements and returns the values
    *
-   * @returns {Array} array of name attributes
+   * @private
+   * @returns {Array} returns elemenrt name and data title
    */
-  const getElemsInView = () => {
+  const findMenuItems = () => {
     const eles = document.querySelectorAll('a[name]');
-    let elesInView = [];
     eles.forEach(element => {
-      const bounding = element.getBoundingClientRect();
-      if (
-        bounding.top >= 0 &&
-        bounding.left >= 0 &&
-        bounding.bottom <=
-          (root.innerHeight || document.documentElement.clientHeight) &&
-        bounding.right <=
-          (root.innerWidth || document.documentElement.clientWidth)
-      ) {
-        elesInView.push(element.getAttribute('name'));
-      }
+      useMenuItems.push({
+        id: element.getAttribute('name'),
+        title: element.getAttribute('data-title'),
+      });
     });
```

Comment:
Move this to outside of `TableOfContents`, above as a private method. You also should not reference `useMenuItems` in here, it should be a standalone function that grabs the elements, and returns back the array. useMenuItems will be defined by either the passed in prop `menuItems`, or will be set as the returned value from the `findMenuItems` method.